delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/model/ValidViewTypes.scala

```diff
@@ -0,0 +1,30 @@
+package ch.epfl.bluebrain.nexus.delta.sourcing.model
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.JsonLdDecoder
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.ValidViewTypes.RestrictedTo
+import io.circe.syntax.EncoderOps
+import io.circe.{Decoder, Encoder, Json}
+
+sealed trait ValidViewTypes {
```

Comment:
To rename
Also as we don't really care if it is an Iri there so we could have a type parameter ?

Reply:
Since it's just incidental that they're iris? Is there a specific example where we could reuse it for a different type? Otherwise that might be over generifying. Considered potentially having different ADTs for schemas / types to make the difference explicit

Reply:
You are right we can keep it like this for now